File: clock_examples/Fortran/time_example.f90
program timing_example
!--------------------------------------------------
! Example of using a timer wrapper in Fortran 2008
!--------------------------------------------------
    use iso_fortran_env, only: real64, stdout => output_unit
    implicit none
    integer           :: sleepy_time = 1
    real(kind=real64) :: time1, time2
    real(kind=real64), external ::  elapsedtime

    time1 = elapsedtime()
    call sleep(sleepy_time)
    time2 = elapsedtime() - time1

    ! The printed time should be only slightly larger than the value of sleepy_time
    write(stdout,666) 'Elapsed time = ', time2, ' seconds'
    666 format(a, es25.17, a)

end program timing_example

File: clock_examples/C++/time_example.c
// 
// Example for calling a timer from C/C++, and using it to time a function. In
// this case it just calls sleep().
//
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>

double elapsedtime(void); // Dumb to have a .h file just for this one line

int main(int argc, char *argv[])
{
    double time1 , time2;
    unsigned int sleepy_time = 1;
    int flag;

    time1 = elapsedtime();
    flag  = sleep(sleepy_time);
    time2 = elapsedtime() - time1;

    // The time printed should be slightly more than sleepy_time
    printf("Elapsed time = %24.17e seconds\n", time2);
    return 0;
};
